runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/Presigner.kt

```diff
@@ -115,7 +115,7 @@ public suspend fun createPresignedRequest(
     serviceConfig: ServicePresignConfig,
     requestConfig: PresignedRequestConfig,
 ): HttpRequest {
-    val givenSigningContext = SigningContext(serviceConfig.serviceId, serviceConfig.region)
+    val givenSigningContext = SigningContext(serviceConfig.signingName, serviceConfig.region)
     val endpoint = serviceConfig.endpointProvider(givenSigningContext)
```

Comment:
**Question**: Why should we pass the _signing name_ to the endpoint resolver? If we don't need service ID here then where do we need it?

Reply:
[This is the line](https://github.com/awslabs/smithy-kotlin/pull/816/files#diff-6de192aa05351178bc825d783da83b16a586059df26763343f277c1bebf326feR129) which made me think we should pass `signingName` to the endpoint resolver. On line 129 we are using the resolved endpoint's `.context?.service` field as a signing name.